### ISSUE TITLE
Report downloads

### ISSUE THREAD

**Issue by elinkronander** [elixir-europe-training/Training-Metrics-Database#66]
**Describe the solution you'd like**

As a Training Coordinator I want to be able to download customized reports, so that I can use them for internal reporting.

**Is your feature request related to a problem? Please describe.**
I used to be able to do this from the old TMD and this is part of my yearly reporting procedure.

**Additional context**

<img width="1219" alt="image" src="https://github.com/user-attachments/assets/5d887da3-4295-4efc-a787-47a16eee0e6e">

**Comment by elinkronander** [comment]
Aggregated reports with more than one question that could be customized would be desired. 

I would like to be able to choose:
- the time-period for the events to include
- which questions to include (granularity here could either be demographics, quality, impact or on individual questions)
- the types of the events to include 
- the target audience of the events to include

**Comment by elinkronander** [comment]
@olisand do you jave any additional comments or insights regarding how SIB used the old TMD for reports?

**Comment by olisand** [comment]
Not really, but in a general mode, I would think the functionality of the so-called "old" TMD regarding reports should ideally be available in the new TMD. See here https://training-metrics-dev.elixir-europe.org/all-reports for examples, with in mind that node-specific reports are only available when logged in (as a node).

**Comment by pdmnyberg** [comment]
I am not sure I understand how this is different from what we currently have at:
- https://tmd.elixir-europe.org/quality
- https://tmd.elixir-europe.org/demographic
- https://tmd.elixir-europe.org/impact

I can see how the printing experience could be improved (It looks kind of terrible). But I am not sure of what exactly you are missing here.

**Comment by elinkronander** [comment]
It might be that I'm just not aware of how to do what I want to do...

What I have managed to do:
Export 1 csv-file for each question by clicking the export button. Then for quality-metrics I need to repeat this for each question and download 6 csv-files.

What I would like to do:
Click 1 button to download a csv-file that includes all the quality-questions (or demographics/impact) with columns named in a similar manner as in the upload template.

**Comment by pdmnyberg** [comment]
> It might be that I'm just not aware of how to do what I want to do. For filtering everything seems to be in place so then it is the "aggregated part" that I'm missing.
> 
> What I have managed to do: Export 1 csv-file for each question by clicking the export button. Then for quality-metrics I need to repeat this for each question and download 6 csv-files and combine them later.
> 
> What I would like to do: Click 1 button to download a csv-file that includes all the quality-questions (or demographics/impact) with columns named in a similar manner as in the upload template.

Thank you. I think this functionality makes a lot of sense formulated this way. Did the old TMD allow for CSV-download? If so, could you provide an example csv-file so that we can try to mimic that format here?

If not I would suggest a structure as follows:
```csv
question,answer,number_of_responses
What is your career stage?,Masters student, 4
What is your career stage?,Other, 50
...
```
Where `What is your career stage?` could be replaced with the question id if that is preferable.

**Comment by elinkronander** [comment]
I don't think the old TMD allowed CSV download (I also haven't used it a lot). 

Your suggested structure seems reasonable to me.

**Comment by olisand** [comment]
I just had a look at the old TMD and I don't find csv export. What is available on the report pages is a "print" button, allowing to print the whole report (all questions, including charts) on a printer or a PDF/postscript document. But this is the same as using your browser File menu.
What you suggest for csv export seems good to me as well.

**Comment by pdmnyberg** [comment]
This should be solved in the development branch now.
